datalad_ria/tests/test_ria_basics.py

```diff
@@ -0,0 +1,137 @@
+    # Doesn't actually create a remote if it fails
+    assert_not_in('ria-remote',
+                  [cfg['name']
+                   for uuid, cfg in ds.repo.get_special_remotes().items()]
+                  )
+    # set up the dataset as well
+    create_ds_in_store(io, store, ds.id, '2', '1')
+    # now should work
+    ds.repo.init_remote('ria-remote', options=init_opts)
+    assert_in('ria-remote',
+              [cfg['name']
+               for uuid, cfg in ds.repo.get_special_remotes().items()]
+              )
+    assert_repo_status(ds.path)
+    # git-annex:remote.log should have:
+    #   - url
+    #   - common_init_opts
+    #   - archive_id (which equals ds id)
+    remote_log = ds.repo.call_git(['cat-file', 'blob', 'git-annex:remote.log'],
+                                  read_only=True)
+    assert_in("url={}".format(url), remote_log)
+    [assert_in(c, remote_log) for c in common_init_opts]
+    assert_in("archive-id={}".format(ds.id), remote_log)
+
+    # re-configure with invalid URL should fail:
+    assert_raises(
+        CommandError,
+        ds.repo.call_annex,
+        ['enableremote', 'ria-remote'] + common_init_opts + [
+            'url=ria+file:///non-existing'])
+    # but re-configure with valid URL should work
+    if has_symlink_capability():
+        link.symlink_to(store)
+        new_url = 'ria+{}'.format(link.as_uri())
+        ds.repo.call_annex(
+            ['enableremote', 'ria-remote'] + common_init_opts + [
+                'url={}'.format(new_url)])
+        # git-annex:remote.log should have:
+        #   - url
+        #   - common_init_opts
+        #   - archive_id (which equals ds id)
+        remote_log = ds.repo.call_git(['cat-file', 'blob',
+                                       'git-annex:remote.log'],
+                                      read_only=True)
+        assert_in("url={}".format(new_url), remote_log)
+        [assert_in(c, remote_log) for c in common_init_opts]
+        assert_in("archive-id={}".format(ds.id), remote_log)
+
+    # we can deal with --sameas, which leads to a special remote not having a
+    # 'name' property, but only a 'sameas-name'. See gh-4259
+    try:
+        ds.repo.init_remote('ora2',
+                            options=init_opts + ['--sameas', 'ria-remote'])
+    except CommandError as e:
+        if 'Invalid option `--sameas' in e.stderr:
+            # annex too old - doesn't know --sameas
+            pass
+        else:
+            raise
+    # TODO: - check output of failures to verify it's failing the right way
+    #       - might require to run initremote directly to get the output
+
+
+def test_initremote_basic_sshurl(ria_sshserver, tmp_path):
+    """Test via SSH"""
+    # retrieve all values from the ssh-server fixture
+    host, port, login, key, path, localpath = ria_sshserver.values()
+    # create all parameters _test_initremote_basic() requires
+    storepath = tmp_path / "store"
+    url = f'ria+ssh://{login}@{host}:{storepath}'
```

Comment:
```suggestion
    url = f'ria+ssh://{login}@{host}:{storepath}'
```

This is not correct. Here is an example URL: `ria+ssh://sshuser@datalad-test-ria:2222/some/path'`

Importantly, the port goes after the `:`, NOT the path.

Inside this will be checked like so:

```
>>> verify_ria_url('ria+ssh://sshuser@datalad-test-ria:2222/some/path', {})
('ssh://sshuser@datalad-test-ria:2222', '/some/path', 'ria+ssh://sshuser@datalad-test-ria:2222/some/path')
```

Reply:
ok, thanks! so this would be correct, right?
```suggestion
    url = f'ria+ssh://{login}@{host}:{port}{storepath}'
```

Reply:
Please see https://github.com/datalad/datalad-ria/pull/61 -- I hope this helps avoid this alltogether.

Reply:
It does now.

Reply:
I'm not sure what you're comment refers to, am I correct to assume that it indicates that we can resolve this conversation now?